app/javascript/App/bundles/CoinShow/HistoricalPriceDataTable.tsx

```diff
@@ -0,0 +1,211 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import {
+  Table,
+  TableHead,
+  TableBody,
+  TableRow,
+  TableCell,
+  Typography,
+  Toolbar,
+} from '@material-ui/core'
+import { withStyles, createStyles } from '@material-ui/core/styles'
+import LoadingIndicator from '../common/components/LoadingIndicator'
+import * as moment from 'moment'
+import { Moment } from 'moment'
+import DateRangeSelect from './DateRangeSelect'
+
+enum STATUSES {
+  INITIALIZING = 'INITIALIZING',
+  LOADING = 'LOADING',
+  READY = 'READY',
+}
+
+interface RawPriceData {
+  coin_key: string
+  to_currency: string
+  time: string
+  open: number
+  high: number
+  low: number
+  close: number
+  volume_from: number // volume in present currency
+  volume_to: number // volume in usd (or to_currency?)
+}
+
+interface PriceData {
+  time: string
+  open: string
+  high: string
+  low: string
+  close: string
+  volume: string
+  formattedTime: string
+  datetime: Moment
+  marketCap: string
+}
+
+interface Props {
+  classes: any
+  initialData?: RawPriceData[]
+  availableSupply: number
+  symbol: string
+}
+
+interface State {
+  data: PriceData[]
+  status: string
+  currency?: string
+  start?: Moment
+  end?: Moment
+}
+
+const styles = (theme) =>
+  createStyles({
+    table: {},
+    toolbar: {},
+    grow: {
+      flexGrow: 1,
+    },
+    picker: {},
+  })
+
+class HistoricalPriceDataTable extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    const initialSortedData = props.initialData
+      ? props.initialData
+          .map<PriceData>(this.parseData.bind(this))
+          .sort(this.sortDataFunc)
+      : undefined
+
+    // Set initial status
+    const statusIsReady = !_.isUndefined(props.initialData)
+    const initialStatus = statusIsReady ? STATUSES.READY : undefined
+    const currency = props.initialData
+      ? _.get(props, ['initialData', '0', 'to_currency'])
+      : undefined
+
+    this.state = {
+      status: initialStatus || STATUSES.INITIALIZING,
+      data: initialSortedData || [],
+      currency,
+    }
+  }
+
+  public sortDataFunc(x: PriceData, y: PriceData) {
+    return y.datetime.diff(x.datetime)
+  }
+
+  public formatPrice(price: number, decimal: number = 2): string {
```

Comment:
Since some cryptocurrencies are valued at sub-cent prices per token, can we go out to 6 decimal places?

While we're at it let's prepend the dollar sign as well, thanks!